src/Orchard.Web/Modules/Orchard.Blogs/Views/Content-BlogPost.SummaryAdmin.cshtml

```diff
@@ -25,17 +28,17 @@
             @T(" | ")
 
             if (contentItem.HasDraft()) {
-                @Html.Link(T("Publish Draft").Text, Url.Action("Publish", "Admin", new { area = "Contents", id = contentItem.Id, returnUrl = Request.ToUrlString() }), new { itemprop = "UnsafeUrl" })
+                @Html.Link(T("Publish Draft").Text, Url.BlogPostPublish(post), new { itemprop = "UnsafeUrl" })
```

Comment:
Correct: using the Contents Admin controller creates the issue, using the BlogPostAdmin controller fixes the issue. This file was added in d31d684 as part of fixing #2575. There have been no other meaningful commits since (that I can tell). 

The permissions checks done in `Content-BlogPost.SummaryAdmin.cshtml` are correctly looking at the Blog Post specific permissions. It is the action links for Publish/Unpublish/Delete that are incorrect, because they forward to Contents Admin Controller actions, and those actions don't check Blog Post specific permissions (but rather, the standard Content permissions).